src/pages/components/PageListPage/PageListSearchAndFilters.tsx

```diff
@@ -0,0 +1,138 @@
+import createFilterHandlers from "@saleor/utils/handlers/filterHandlers";
+import { mapEdgesToItems } from "@saleor/utils/maps";
+import React from "react";
+import { useIntl } from "react-intl";
+
+import {
+  createFilterStructure,
+  deleteFilterTab,
+  getActiveFilters,
+  getFilterOpts,
+  getFilterQueryParam,
+  getFiltersCurrentTab,
+  getFilterTabs,
+  saveFilterTab
+} from "./filters";
+import { pagesListSearchAndFiltersMessages as messages } from "./messages";
+import { PageListActionDialogOpts } from "./PageListPage";
+
+interface PageListSearchAndFiltersProps {
+  params: PageListUrlQueryParams;
+  actionDialogOpts: PageListActionDialogOpts;
+}
+
+const PageListSearchAndFilters: React.FC<PageListSearchAndFiltersProps> = ({
+  params,
+  actionDialogOpts
+}) => {
+  const navigate = useNavigator();
+  const intl = useIntl();
+
+  const defaultSearchVariables = {
+    variables: {
+      ...DEFAULT_INITIAL_SEARCH_DATA,
+      first: 5
+    }
+  };
+
+  const { reset } = useBulkActions(params.ids);
+
+  const {
+    loadMore: fetchMorePageTypes,
+    search: searchPageTypes,
+    result: searchPageTypesResult
+  } = usePageTypeSearch(defaultSearchVariables);
+
+  const filterOpts = getFilterOpts({
+    params,
+    pageTypes: mapEdgesToItems(searchPageTypesResult?.data?.search),
+    pageTypesProps: {
+      ...getSearchFetchMoreProps(searchPageTypesResult, fetchMorePageTypes),
+      onSearchChange: searchPageTypes
+    }
+  });
+
+  const [
+    changeFilters,
+    resetFilters,
+    handleSearchChange
+  ] = createFilterHandlers({
+    createUrl: pageListUrl,
+    getFilterQueryParam,
+    navigate,
+    params,
+    cleanupFn: reset
+  });
+
+  const filterStrucutre = createFilterStructure(intl, filterOpts);
+
+  const { open: openModal, close: closeModal } = actionDialogOpts;
+
+  const handleTabChange = (tab: number) => {
+    navigate(
+      pageListUrl({
+        activeTab: tab.toString(),
+        ...getFilterTabs()[tab - 1].data
+      })
+    );
+  };
+
+  const tabs = getFilterTabs();
+  const currentTab = getFiltersCurrentTab(params, tabs);
+
+  const handleTabSave = (data: SaveFilterTabDialogFormData) => {
+    saveFilterTab(data.name, getActiveFilters(params));
+    handleTabChange(tabs.length + 1);
+  };
+
+  const handleTabDelete = () => {
+    deleteFilterTab(currentTab);
+    reset();
+    navigate(pageListUrl());
+  };
+
+  return (
+    <>
+      <FilterBar
+        filterStructure={filterStrucutre}
+        initialSearch={""}
+        onAll={resetFilters}
+        onFilterChange={changeFilters}
+        onSearchChange={handleSearchChange}
+        searchPlaceholder={intl.formatMessage(messages.searchPlaceholder)}
+        allTabLabel={"All Pages"}
+        tabs={tabs.map(({ name }) => name)}
+        currentTab={currentTab}
+        onTabDelete={handleTabDelete}
+        onTabChange={handleTabChange}
+        onTabSave={() => openModal("save-search")}
+      />
+      <SaveFilterTabDialog
```

Comment:
Its not consistent with the rest of the project, where we keep dialog outside filtr component.

Reply:
@dominik-zeglen Gift Cards has been done like that - https://github.com/saleor/saleor-dashboard/blob/main/src/giftCards/GiftCardsList/GiftCardListSearchAndFilters/GiftCardListSearchAndFilters.tsx

Reply:
Yes, inconsistency is the downside. Patterns used in gift card development were experimental and we did not reflect on this yet.